src/IndexManagement/IndexManager.php

```diff
@@ -236,6 +236,32 @@ function () use ($source, $destination) {
         );
     }
 
+    public function putSettings(string $index, array $settings = []): void
+    {
+        $body = ['index' => []];
+
+        foreach ($settings as $key => $setting) {
+            if ($key === 'refresh_interval') {
+                $body['index'][$key] = $setting;
+            }
+            if ($key === 'number_of_replicas') {
+                $body['index'][$key] = $setting;
+            }
+        }
+
+        $this->execute(
+            function () use ($index, $body) {
+                return $this->client->indices()->putSettings([
+                    'index' => $index,
+                    'body' => $body,
+                ]);
+            },
+            true,
+            'Put Settings succeed.',
```

Comment:
that is the log message in case of failure, so you will want to rephrase that ;)

Reply:
:smile: "Task failed successfully!"